include/seqan3/core/algorithm/all.hpp

```diff
@@ -38,13 +38,68 @@
  */
 
 #pragma once
-//!\cond DEV
+
 /*!\defgroup algorithm Algorithm
  * \ingroup core
  * \brief Contains core functionality used to configure algorithms.
- * \todo Write detailed description about usage of this module.
+ *
+ * \details
+ *
+ * In SeqAn there are many, e.g. alignment or search interfaces, that can be configured through
+ * many different settings and policies that alter the execution of the respective algorithm.
+ * These configurations can be orthogonal or might be mutually exclusive and can make interfaces very difficult to use.
+ * This module provides a basic system to manage the configurations of algorithms using a unified interface.
+ *
+ * ### Usage
+ *
+ * The basis of any algorithm configuration are configuration elements. These are objects that handle a specific
+ * setting and must satisfy the seqan3::detail::config_element_concept. The following snippet demonstrates a
+ * basic setup for such configuration elements.
+ *
+ * \snippet test/snippet/core/algorithm/configuration.cpp configuration_setup
+ *
+ * Here, two classes with the name `bar` and `foo` are created. They can be normal or template classes and must
+ * inherit from seqan3::pipeable_config_element and contain a member with the name `value` to satisfy the respective
+ * concept. The separate `value` member is used for a proper encapsulation from the actual setting parameter.
+ * For example the alignment algorithms require a scoring scheme, but the scoring scheme itself should not be pipeable
```

Comment:
> The separate `value` member is used for a proper encapsulation from the actual setting parameter. For example the alignment algorithms require a scoring scheme, but the scoring scheme itself should not be pipeable with other settings.

I do not understand what you are trying to tell here... what is a setting parameter? Is the scoring scheme a configuration element? This is very specific and does not blend in well the explanations of what is going on so far.